Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectNavigation.cs

```diff
@@ -185,6 +180,48 @@ public override TokenParser ProvisionObjects(Web web, ProvisioningTemplate templ
 
         #region Utility methods
 
+        private bool WebSupportsProvisionNavigation(Web web, ProvisioningTemplate template)
+        {
+            if (template.Navigation == null) return true;
+
+            bool isNavSupported = true;
+            // The Navigation handler for managed metedata only works for sites with Publishing Features enabled
+            if (!web.IsPublishingWeb())
+            {
+                if (template.Navigation.GlobalNavigation != null
+                    && template.Navigation.GlobalNavigation.NavigationType == GlobalNavigationType.Managed)
+                {
```

Comment:
Hi,
I did not test this scenario. Which means if you have a template with managed nav for global, and structured for quick, then it will skip both if trying to apply to a non-pub site, which should be ok. Same the other way around.

The check will not apply any navigation to a non-pub site if the template has managed nav in it. I think this is sensible as the logic to apply just one type would complicate the code and logic as you would need to check all cases in ProvisionObjects - sort of breaking the logic of WillProvision. Or the navigation would need to be split into different classes for each type of navigation(?).